File: tests/fixtures/skeleton/go/module.go
package service

import (
	"fmt"
	"sync"

	"github.com/org/app/internal/db"
	"github.com/org/app/internal/models"
)

type Repository struct {
	mu     sync.Mutex
	session *db.Session
	items  map[string]*models.User
}

func (r *Repository) Find(pk string) (*models.User, error) {
	r.mu.Lock()
	defer r.mu.Unlock()
	return r.items[pk], nil
}

func (r *Repository) Save(u *models.User) error {
	r.mu.Lock()
	defer r.mu.Unlock()
	r.items[u.ID] = u
	return nil
}

type Greeter interface {
	Greet() string
}

func NewRepository(s *db.Session) *Repository {
	return &Repository{session: s, items: make(map[string]*models.User)}
}

var _ = func(x int) int {
	return x * 2
}

File: tests/fixtures/skeleton/rs/module.rs
use std::collections::HashMap;
use std::sync::Arc;

use crate::db::Session;
use crate::models::{Model, User};

pub struct Repository<T>
where
    T: Model,
{
    session: Arc<Session>,
    cache: HashMap<String, T>,
}

impl<T: Model> Repository<T> {
    pub fn new(session: Arc<Session>) -> Self {
        Self {
            session,
            cache: HashMap::new(),
        }
    }

    pub fn find(&self, pk: &str) -> Option<T> {
        self.cache.get(pk).cloned()
    }

    pub fn save(&mut self, obj: T) {
        let key = obj.id().to_string();
        self.cache.insert(key, obj);
    }
}

#[derive(Debug, Clone)]
pub enum Status {
    Active,
    Inactive,
}

pub trait RepositoryTrait {
    fn find(&self, pk: &str) -> Option<T>;
    fn save(&mut self, obj: T) {
        let _ = obj;
    }
}

pub fn lookup(id: &str) -> Result<Status, Box<dyn std::error::Error>> {
    Ok(Status::Active)
}
